openedx/core/djangoapps/credentials/utils.py

```diff
@@ -0,0 +1,69 @@
+"""Helper functions for working with Credentials."""
+from __future__ import unicode_literals
+import logging
+
+from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
+from openedx.core.djangoapps.programs.utils import get_programs_for_credentials
+from openedx.core.lib.api_utils import get_api_data
+
+
+log = logging.getLogger(__name__)
+
+
+def get_user_credentials(user):
+    """Given a user, get credentials earned from the Credentials service.
+    Arguments:
+        user (User): The user to authenticate as when requesting credentials.
+    Returns:
+        list of dict, representing credentials returned by the Credentials
+        service.
+    """
+    credential_configuration = CredentialsApiConfig.current()
+    user_query = {'username': user.username}
+    # Bypass caching for staff users, who may be generating credentials and
+    # want to see them displayed immediately.
+    use_cache = credential_configuration.is_cache_enabled and not user.is_staff
+    cache_key = None
+    if use_cache:
+        cache_key = credential_configuration.CACHE_KEY + '.' + user.username
```

Comment:
You can simplify this by using the presence of the `cache_key` to determine whether or not the cache should be used. The `use_cache` variable is redundant. If caching is disabled or the user is staff, set the cache key to None.